tutorials/cassandraInput/src/test/java/com/datatorrent/cassandra/CassandraApplicationTest.java

```diff
@@ -0,0 +1,146 @@
+      String createTable = "CREATE TABLE IF NOT EXISTS "
+          + KEYSPACE
+          + "."
+          + TABLE_NAME
+          + " (id uuid PRIMARY KEY,age int,lastname text,test boolean,floatvalue float,doubleValue double,set1 set<int>,list1 list<int>,map1 map<text,int>,last_visited timestamp);";
+      session.execute(createTable);
+      insertEventsInTable(10);
+    } catch (Throwable e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+  @AfterClass
+  public static void cleanup()
+  {
+    if (session != null) {
+      session.execute("DROP TABLE " + CassandraTransactionalStore.DEFAULT_META_TABLE);
+      session.execute("DROP TABLE " + KEYSPACE + "." + TABLE_NAME);
+      session.close();
+    }
+    if (cluster != null) {
+      cluster.close();
+    }
+  }
+
+  private static void insertEventsInTable(int numEvents)
+  {
+    try {
+      Cluster cluster = Cluster.builder().addContactPoint(NODE).build();
+      Session session = cluster.connect(KEYSPACE);
+
+      String insert = "INSERT INTO " + KEYSPACE + "." + TABLE_NAME + " (ID,lastname,age)" + " VALUES (?,?,?);";
+      PreparedStatement stmt = session.prepare(insert);
+      BoundStatement boundStatement = new BoundStatement(stmt);
+      for (int i = 0; i < numEvents; i++) {
+        ids.add(UUID.randomUUID());
+        mapNames.put(i, "test" + i);
+        mapAge.put(i, i + 10);
+        session.execute(boundStatement.bind(ids.get(i), "test" + i, i + 10));
+      }
+    } catch (DriverException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testApplication() throws IOException, Exception
+  {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      conf.addResource(this.getClass().getResourceAsStream("/META-INF/properties-CassandraInputApplication.xml"));
+      conf.set("dt.operator.CassandraReader.prop.store.node", "localhost");
+      conf.set("dt.operator.CassandraReader.prop.store.keyspace", KEYSPACE);
+      conf.set("dt.operator.CassandraReader.prop.tablename", TABLE_NAME);
+      lma.prepareDAG(new Application(), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.run(10000); // runs for 10 seconds and quits
+
+      ((StramLocalCluster)lc).setExitCondition(new Callable<Boolean>()
+      {
+        @Override
+        public Boolean call() throws Exception
+        {
+          if (getNumOfEventsInStore() == 10) {
+            return true;
+          }
+          return false;
+        }
+      });
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  private long getNumOfEventsInStore()
+  {
+    String countQuery = "SELECT count(*) from " + KEYSPACE + "." + TABLE_NAME + ";";
+    ResultSet resultSetCount = session.execute(countQuery);
+    for (Row row : resultSetCount) {
+      return row.getLong(0);
+    }
+    return 0;
+
+  }
+}
```

Comment:
I tried using "cassandra-unit" embedded server. It seems difficult to use it as of now as there is no way to inject, cluster or session from outside. Also using mocks in functional/application test doesn't look like a good idea. I have already created a ticket to use embedded cassandra server in Malhar. Once we do that (that may involve some code changes like exposing some getter/setters etc.) we can write tests with embedded server in examples as well.